src/context.ts

```diff
@@ -27,8 +30,20 @@ function getUserId(token?: string): string {
       throw new ApolloError(err.name === 'TokenExpiredError' ? 'Token expired' : 'Invalid authentication token');
     }
     return decoded;
-  }) as unknown as { userId: string };
-  return verificationTokenResult.userId;
+  }) as unknown as { sessionId: string };
+  const { sessionId } = verificationTokenResult;
+  const session = await xprisma.session.findUniqueOrThrow({
+    where: {
+      id: sessionId,
+    },
+    include: {
+      creator: true,
+    },
+  });
+  if (session.revokedAt && session.revokedAt < new Date()) {
+    throw new ApolloError('Session expired');
+  }
+  return session.creator.id;
```

Comment:
Let's return user object and rename `getUserId` to `getUser` for clarity (as suggested in the issue description)